### ISSUE TITLE
Improved error messages: Categorize existing check failures and trace messages 

### ISSUE THREAD

**Issue by erohmensing** [airbytehq/airbyte#24618]

**Comment by nataliekwong** [comment]
@letiescanciano I see `failure_reason` in the FE events, and for this event it was `Failed to run connection tests.` In the UI it shows a different permissions error. It sounds like we're tracking both on the BE - could we do similarly on the FE?

<img width="406" alt="Screenshot 2023-03-27 at 10 39 15 PM" src="https://user-images.githubusercontent.com/38087517/228138858-1e67a2ab-f6f5-4b17-9a0a-f5298ec2c569.png">

**Comment by erohmensing** [comment]
Yes, we can - I think I put a note on Leti's PR about this. I've put this in my backlog, I'll let you know when its in :) https://app.zenhub.com/workspaces/connector-operations-6318ede5eda32e0010ef8566/issues/gh/airbytehq/airbyte/24748

**Comment by nataliekwong** [comment]
Ah thank you! Appreciate it @erohmensing

**Comment by erohmensing** [comment]
The above PR has been merged for a while, so we should have data to look into!
